common/Services/NotificationService.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.Input;
+using CommunityToolkit.WinUI;
+using CommunityToolkit.WinUI.Behaviors;
+using DevHome.Common.Contracts;
+using DevHome.Common.Environments.Helpers;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.Windows.AppLifecycle;
+using Microsoft.Windows.AppNotifications;
+using Microsoft.Windows.AppNotifications.Builder;
+using Serilog;
+using Windows.ApplicationModel.Activation;
+
+namespace DevHome.Common.Services;
+
+public class NotificationService
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(NotificationService));
+
+    private readonly IWindowsIdentityService _windowsIdentityService;
+
+    private readonly string _componentName = "NotificationService";
+
+    private readonly string _hyperVText = "Hyper-V";
+
+    private readonly string _microsoftText = "Microsoft";
+
+    private readonly StringResource _stringResource;
+
+    private readonly Microsoft.UI.Dispatching.DispatcherQueue _dispatcher;
+
+    private StackedNotificationsBehavior? _notificationQueue;
+
+    public NotificationService(IWindowsIdentityService windowsIdentityService)
+    {
+        _windowsIdentityService = windowsIdentityService;
+        _dispatcher = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
```

Comment:
If this isn't running on the UI thread, it will return null. Please use WindowEx.DispatcherQueue instead.

Reply:
It is running. Plus, I used the pattern from other files. :)

Reply:
I started an internal thread, you're on it. I believe this will indeed intermittently crash when this happens to run off the UI thread (in my experience, more often off the debugger than under it). Opened https://github.com/microsoft/devhome/issues/2504 to fix it.